### PR TITLE
Chore/sync env vars for gateway

### LINKED ISSUE #29: Sync env vars for gateway
# Description
As a developer, I want clear documentation regarding the available options for configuring a service via the environment. When multiple related services each have environment variables that service the same purpose, they should be consistently named. A sample .env.template file or README documentation should document all available environment variables, their purpose, range of values, and whether required or optional.

# Acceptance Criteria
- [x] Replace ENVIRONMENT with standard variable name to enable dev environment features
- [x] Change `WEB_HOOK_POST_MAX_RETRIES` to `WEBHOOK_FAILURE_THRESHOLD` to match standard
- [x] Remove references to unused environment: `HEALTH_CHECK_*`
- [x] Add support for `WEBHOOK_RETRY_INTERVAL`?
- Ensure all variables are properly documented in:
    - [x] ENVIRONMENT.md
    - [x]  .env.template

### PR DESCRIPTION
**Pull request by JoeCap08055**
This PR removes unused environment/config, renames env vars to be consistent with other services, and adds documentation for all environment config options.

Closes #29

### CODE REVIEW

File: ENVIRONMENT.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+|`FREQUENCY_URL`                           | Blockchain node address                                                                                                                                                                                                  |       http(s): or ws(s): URL       |      Y       |         |
+|`IPFS_BASIC_AUTH_SECRET`|If using Infura, put auth token here, or leave blank for Kubo RPC|string|N|blank|
+|`IPFS_BASIC_AUTH_USER`|If using Infura, put Project ID here, or leave blank for Kubo RPC|string|N|blank|
+|`IPFS_ENDPOINT`|URL to IPFS endpoint|URL|Y||
+|`IPFS_GATEWAY_URL`|IPFS gateway URL. '[CID]' is a token that will be replaced with an actual content ID|URL template|Y||
+|`QUEUE_HIGH_WATER`                        | Max number of jobs allowed on the 'graphUpdateQueue' before blockchain scan will be paused to allow queue to drain                                                                                                       |               >= 100               |              |  1000   |
```
Comment (saraswatpuneet) [inline ENVIRONMENT.md:14]: not sure this is needed in content watcher, you can delete `QUEUE_HIGH_WATER`

File: ENVIRONMENT.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+|`REDIS_URL`                               | Connection URL for Redis                                                                                                                                                                                                 |                URL                 |      Y       |
+|`STARTING_BLOCK`|Block number from which the service will start scanning the chain|> 0||1|
+|`WEBHOOK_FAILURE_THRESHOLD`               | Number of failures allowing in the provider webhook before the service is marked down                                                                                                                                    |                > 0                 |              |    3    |
+|`WEBHOOK_RETRY_INTERVAL_SECONDS`          | Number of seconds between provider webhook retry attempts when failing                                                                                                                                                   |                > 0                 |              |   10    |
```
Comment (saraswatpuneet) [inline ENVIRONMENT.md:18]: Same with `WEBHOOK_FAILURE_THRESHOLD` and `WEBHOOK_RETRY_INTERVAL_SECONDS` not sure if they are being used in watcher

Comment (saraswatpuneet) [reply ENVIRONMENT.md:18]: I guess they are being used when sending out notifications to webhooks

File: libs/common/src/pubsub/pubsub.service.ts [hunk 3]
```diff
@@ -55,6 +56,10 @@ export class PubSubService {
             this.logger.error(`Failed to send announcement to webhook: ${webhookUrl}`);
             this.logger.error(error);
             retries += 1;
+            // eslint-disable-next-line no-await-in-loop
+            await new Promise((r) => {
```
Comment (enddynayn) [inline libs/common/src/pubsub/pubsub.service.ts:60]: seems a bit out of scope. I think it might be a good idea to separate this and write a test for it.

**Review by enddynayn — APPROVED** [review]
👍 LGTM

File: ENVIRONMENT.md (hunk 1 above)
Comment (JoeCap08055) [reply ENVIRONMENT.md:14]: Confirmed it's used [here](https://github.com/AmplicaLabs/content-watcher-service/blob/6a9f8663e5f2e2d4ff3a715473c79365a8f84183/libs/common/src/ipfs/ipfs.dsnp.ts#L207) and [here](https://github.com/AmplicaLabs/content-watcher-service/blob/6a9f8663e5f2e2d4ff3a715473c79365a8f84183/libs/common/src/scanner/scanner.ts#L131)

Comment (saraswatpuneet) [reply ENVIRONMENT.md:14]: Then we can change the description, it currently says this is related to graphUpdateQueue

File: libs/common/src/pubsub/pubsub.service.ts (hunk 3 above)
Comment (JoeCap08055) [reply libs/common/src/pubsub/pubsub.service.ts:60]: Part of the purpose of this exercise is to make all of our gateway services configurable in the same standard ways. Our other gateway services that interact with webhooks all honor this environment variable; rather than delete it as unused, it seemed logical to add in support for the delay. Plus hammering a failed webhook with no delay between retries is not good practice.

Additional considerations such as exponential backoff, etc, would be more complex and can be deferred to a separate issue, but since this is a one-liner, it seems appropriate.

Comment (enddynayn) [reply libs/common/src/pubsub/pubsub.service.ts:60]: Yeah. I do agree that it's needed. I think one of the concerns that has been brought during our retro was adding test. I think this one of those cases that we can be proactive to ensure the quality of our code. I do recommend adding a ticket for this so that it does not go unnoticed. Please try to keep things scoped to your ticket.
